internal/datanode/compactor.go

```diff
@@ -466,17 +466,23 @@ func (t *compactionTask) compact() error {
 	}
 
 	uploadStart := time.Now()
-	cpaths, err := t.upload(ctxTimeout, targetSegID, partID, iDatas, deltaBuf.delData, meta)
-	if err != nil {
-		log.Error("compact wrong", zap.Int64("planID", t.plan.GetPlanID()), zap.Error(err))
-		return err
+	var segPaths *cpaths
+	if numRows <= 0 {
+		segPaths, err = t.upload(ctxTimeout, targetSegID, partID, iDatas, deltaBuf.delData, meta)
+		if err != nil {
+			log.Error("compact wrong", zap.Int64("planID", t.plan.GetPlanID()), zap.Error(err))
+			return err
+		}
```

Comment:
I don't see the point of this, `upload` already takes care of empty segment cases https://github.com/milvus-io/milvus/blob/b0923f1299fea669380456e8175f971576d244af/internal/datanode/binlog_io.go#L125

Reply:
it just warns that you are trying to upload empty data, but still creates the blob